src/vs/editor/common/editorCommon.ts

```diff
@@ -3801,6 +3810,13 @@ export interface ICommonCodeEditor extends IEditor {
 	onDidType(listener: (text: string) => void): IDisposable;
 
 	/**
+	 * An event emitted when users paste text in the editor.
+	 * @event
+	 * @internal
+	 */
+	onDidPaste(listener: (range: Range) => void): IDisposable;
```

Comment:
Here we say we give out a `Range`, but the implementation gives out an `IRange`. Please adjust the implementation.